### PR TITLE
[scope] - [development] implemented security setting in post event.

### PR DESCRIPTION
**Pull request by kodeerkarthik**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -5,7 +5,8 @@ class PostsController < ApplicationController
     skip_before_action :verify_authenticity_token
 
   def index
-    @posts = Post.order('created_at DESC')
+    @post = Post.where(:User_id => current_user.id).notpublished
```
Comment (Sujina-Saddival) [inline app/controllers/posts_controller.rb]: arrays will be returned - so change this to posts

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -5,7 +5,8 @@ class PostsController < ApplicationController
     skip_before_action :verify_authenticity_token
 
   def index
-    @posts = Post.order('created_at DESC')
+    @post = Post.where(:User_id => current_user.id).notpublished  
+    @posts = Post.published
```
Comment (Sujina-Saddival) [inline app/controllers/posts_controller.rb]: change this to @published_posts

File: app/models/ability.rb [hunk 3]
```diff
@@ -18,8 +18,10 @@ def initialize(user)
         can :manage, :all
       else
         can :read, Article
+        can :read, Post
         can :update, Article if user.has_role?(:manager, Article)
         can :update, Article if user.has_role?(:moderator, Article)
+        can :manage, Post, User_id: user.id
```
Comment (Sujina-Saddival) [inline app/models/ability.rb]: User_id? camel case?

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -5,6 +5,7 @@
             <%= link_to 'Add new Image', new_post_path, id:'new_path', remote:true, :class=>"btn btn-sm btn-warning" %>
             <hr>
             <%= render @posts %>
+            <%= render @post %>
```
Comment (Sujina-Saddival) [inline app/views/posts/index.html.erb]: change this. give understandable variable names

File: app/models/ability.rb (hunk 3 above)
Comment (kodeerkarthik) [reply app/models/ability.rb]: User_id is the field belongs to the Post table. so it has to be used the same as User_id.

Comment (kodeerkarthik) [reply app/models/ability.rb]: If I want to change this, I should change in the Post table migration.

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (kodeerkarthik) [reply app/controllers/posts_controller.rb]: I have replaced @post with @posts

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (kodeerkarthik) [reply app/controllers/posts_controller.rb]: I have replaced @post with @posts and @posts with @published_posts

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (kodeerkarthik) [reply app/views/posts/index.html.erb]: changed

File: app/models/ability.rb (hunk 3 above)
Comment (Sujina-Saddival) [reply app/models/ability.rb]: it should be **u**ser_id not **U**ser_id

**Pull request merged**